### DISCUSSION TITLE
Problem on submit_unseal_keys()

### DISCUSSION THREAD

**Discussion by Manfr-edi** [hvac/hvac#1065] [Q&A]
Hi everyone,

I have an issue on the function for unsealing the vault.
When i call the submit function, the vault reports a log with a "post-unseal setup complete" message, but the python script crashes and give the following error on first access to the vault:
 local node not active but active cluster node not found, on get http://192.168.125.55:8200/v1/kv/data/dbsecrets

after this crash, the vault is unsealed, so if i re-run the script, everything is ok.

This is the code i used, the script breakes in the first read_secret_version.
```python
client = hvac.Client(url=os.environ['VAULT_URL'], token=os.environ['VAULT_TOKEN'])
keys = [os.environ['VAULT_UNSEAL_KEY_1'], os.environ['VAULT_UNSEAL_KEY_2'], os.environ['VAULT_UNSEAL_KEY_3']]
if client.sys.is_sealed():
    client.sys.submit_unseal_keys(keys)
postgres_admin = client.secrets.kv.read_secret_version(mount_point='kv', path='dbsecrets')['data']['data']['DB_USERNAME']
postgres_pwd = client.secrets.kv.read_secret_version(mount_point='kv', path='dbsecrets')['data']['data']['DB_PASSWORD']
```

![error](https://user-images.githubusercontent.com/86465812/200278066-0b5e8011-0df5-4a85-b518-a99255879716.png)

**Comment by briantist** [comment]
Hi @Manfr-edi , this is likely an issue with [Vault Eventual Consistency](https://developer.hashicorp.com/vault/docs/enterprise/consistency). It's going to be best to first understand your Vault cluster setup, and take a look at the mitigations in that document.

On the `hvac` side, right now the main thing we recommend for dealing with some of that is retries, see:
- https://hvac.readthedocs.io/en/stable/advanced_usage.html#retrying-failed-requests
